src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const eachAction = [];
+  let initialState = { ...state };
+
+  for (const action of actions) {
+    let newState = { ...initialState };
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
+    }
```

Comment:
[CODE STYLE]: The `switch` statement should have a `default` case to handle unexpected action types. This is important for error handling and to ensure that the function behaves predictably even when encountering unknown action types.